dataland-frontend/tests/component/components/forms/UploadFieldSpecificReports.cy.ts

```diff
@@ -0,0 +1,176 @@
+    let createDataset;
+    let dataType: DataTypeEnum;
+    if (framework == "sfdr") {
+      dataType = DataTypeEnum.Sfdr;
+      createDataset = CreateSfdrDataset;
+    } else {
+      dataType = DataTypeEnum.P2p;
+      createDataset = CreateP2pDataset;
+    }
+
+    cy.intercept(`**/documents/*`, cy.spy().as("documentExists"));
+    cy.intercept("POST", `/api/data/${dataType}`, {
+      statusCode: 200,
+    });
+    cy.mountWithPlugins(createDataset, {
+      keycloak: minimalKeycloakMock({}),
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      props: {
+        companyID: companyId,
+      },
+    });
+  }
+
+  it("Check if the document uploads in Sfdr upload page do not interfere", () => {
+    console.log(hashForFileWithOneByteSize);
+    const setOfHashesThatShouldBeCheckedForExistence = new Set([
+      hashForFileWithOneByteSize,
+      hashForFileWithTwoBytesSize,
+      hashForFileWithThreeBytesSize,
+    ]);
+    setOfHashesThatShouldBeCheckedForExistence.forEach((hash) => {
+      interceptEachUpload(hash);
+    });
+    mountPluginAndInterceptUploads("sfdr");
+    createSfdrDataset.fillRequiredFields();
+    uploadAndReferenceSfdrReferencedReport("Sfdr1", 1);
+    uploadFieldSpecificDocuments("Sfdr2", 2, "sustainableAgriculturePolicy");
+    uploadFieldSpecificDocuments("Sfdr3", 3, "sustainableOceansAndSeasPolicy");
+    uploadFieldSpecificDocuments("Sfdr4", 1, "environmentalPolicy");
+    cy.wait(100);
+    submitButton.buttonAppearsEnabled();
+    submitButton.clickButton();
+
+    setOfHashesThatShouldBeCheckedForExistence.forEach((hash) => {
+      cy.wait(`@documentExists-${hash}`);
+    });
+    cy.wait(100);
+    cy.get("@documentExists").should("have.been.calledThrice");
+  });
+
+  it("Check if the document uploads in Sfdr upload page still work properly if some document got removed or replaced", () => {
+    console.log(hashForFileWithOneByteSize);
+    const setOfHashesThatShouldBeCheckedForExistence = new Set([hashForFileWithTwoBytesSize]);
+    setOfHashesThatShouldBeCheckedForExistence.forEach((hash) => {
+      interceptEachUpload(hash);
+    });
+    mountPluginAndInterceptUploads("sfdr");
+    createSfdrDataset.fillRequiredFields();
+    uploadFieldSpecificDocuments("first", 1, "sustainableAgriculturePolicy");
+    cy.wait(100);
+    cy.get('div[data-test="sustainableAgriculturePolicy"] button .pi-times').click();
+    uploadFieldSpecificDocuments("second", 2, "sustainableAgriculturePolicy");
+    uploadFieldSpecificDocuments("fourth", 3, "environmentalPolicy");
+    cy.wait(100);
+    cy.get('div[data-test="environmentalPolicy"] button .pi-times').click();
+    cy.wait(100);
+    submitButton.buttonAppearsEnabled();
+    submitButton.clickButton();
+
+    setOfHashesThatShouldBeCheckedForExistence.forEach((hash) => {
+      cy.wait(`@documentExists-${hash}`);
+    });
+    cy.wait(100);
+    cy.get("@documentExists").should("have.been.calledOnce");
```

Comment:
Because one "first" gets replaced by "second" and "third" gets removed entirely. The reason for this test is to make sure that file deletion works properly.

Reply:
ok